packages/core/content-manager/server/src/services/configuration.ts

```diff
@@ -1,37 +1,48 @@
 import { intersection, difference } from 'lodash';
+
+import type { Configuration } from '../../../shared/contracts/content-types';
+
 import { createDefaultConfiguration, syncConfiguration } from './utils/configuration';
 
-export default ({ isComponent, prefix, storeUtils, getModels }: any) => {
-  const uidToStoreKey = (uid: any) => {
+export type ConfigurationUpdate = Omit<Configuration, 'uid'> & {
+  options?: Record<string, unknown>;
+};
```

Comment:
dont have other alternatives, but sharing the type between shared and here makes me feel a bit uncomfortable.
And even more when the shared definition is different from what its needed here 🤔

Reply:
I think in cases like this I would redefine the type here, if it's not a masssive one. Just because in this case it's not a 1:1 where each type is the same.

Reply:
Okay makes sense, it seems we might need that across multiple services. e.g.

services/components.ts
services/content-types.ts
services/metrics.ts
services/utils/configuration/layouts.ts
etc

Do you think we should define the type in configuration services and import it from elsewhere?

Reply:
you mean all of those services use the configuration type?
Are the configuration type in components, contenttypes metrics, ... the same?  or they are different?

I would have a type definition in services, and another one in the shared folder.
But if we see fit, I would define the type in services and import it from the shared folder

Reply:
what do you think of the changes made in https://github.com/strapi/strapi/pull/18869/commits/88a50d628c8d837ecffd1f0da4635980b6b9b47c ?

This type matches the usage in components and content types services, for the rest of the services I can reuse this if it matches or redefine types otherwise. Should I redefine `Settings, Metadatas & Layouts` to keep it separate?

Reply:
ah! This looks much better indeed.
LGTM 🚀